### ISSUE TITLE
Including = symbol in JSON post body causes body to become truncated

### ISSUE THREAD

**Issue by andypiper** [twitter/twurl#77]
Currently, including the = symbol inside any JSON body posted to the API via twurl causes the body content to be truncated at the = symbol, and the request will subsequently fail. This occurs even with the Content-Type set to application/json.

**Comment by andypiper** [comment]
I'm seeing that this works if the = is encoded as %3d.

**Comment by smaeda-ks** [comment]
I caught this issue today as well. For those who are curious, the problem is coming from this logic:
https://github.com/twitter/twurl/blob/7eb9a1364cbc9e8ee30a81cf36db384eb363e0af/lib/twurl/cli.rb#L234

It looks like twurl is doing this by _default_ and there's no way to skip this which is very confusing and doesn't make sense to me having this logic for all POST data. 

> This occurs even with the Content-Type set to application/json.

Exactly... curl doesn't do anything like this.
https://ec.haxx.se/http-post.html

Probably, it's difficult to just remove this logic as it will potentially require a bunch of documentation fix (on [developer.twitter.com](https://developer.twitter.com/)) and other critical stuff, but we should have a logic to check if a `Content-Type` is set by option (`-H`) and if it _contains_ `application/json` in its value, so that we could skip this unnessesary parse for json POST body at least.

I also noticed there's a `-r, --raw-data [data] Sends the specified data as it is in a POST request to the HTTP server.` option:
https://github.com/twitter/twurl/blob/7eb9a1364cbc9e8ee30a81cf36db384eb363e0af/lib/twurl/cli.rb#L240-L246

but as you can guess it doesn't work because of the parse, again...

Well, there's a logic that supposed to pass a "raw" body data (was introduced in https://github.com/twitter/twurl/pull/60):
https://github.com/twitter/twurl/blob/bac93af79c5afb771d10de375a754f67626a8c31/lib/twurl/oauth_client.rb#L111-L112

but it only gets a "key" even though rest of the text after `=` parsed by above logics is in its value.

**Comment by smaeda-ks** [comment]
FYI, I looked at the code further (I'm Ruby beginner though..) and noticed that this fix isn't that easy as I thought initially. Since twurl is using OptionParser in order to parse command-line args, it reads args one by one with an actual input order. That means, if a user set options like this oder:

`twurl -d '{"data": "foo=bar"}' -A 'Content-Type: application/json' '/path/to/endpoint'`

`data` function doesn't know if there's a `Content-Type` header in `options.headers` or not at this stage.
https://github.com/twitter/twurl/blob/7eb9a1364cbc9e8ee30a81cf36db384eb363e0af/lib/twurl/cli.rb#L231-L238

Maybe we can do pattern matching using `ARGV` or something, but not sure if it's a reasonable way. So the easiet wasy I tested is to modify this block:
https://github.com/twitter/twurl/blob/bac93af79c5afb771d10de375a754f67626a8c31/lib/twurl/oauth_client.rb#L111-L112

to something like this:

```ruby
elsif request.content_type && options.data
  data = ''

  if options.data.length == 1 && options.data.values.first == nil
    data += options.data.keys.first
  else
    options.data.each_with_index do |(key, value), i|
      data += (i == 0 ? '' : '&') + key.to_s + (value == nil ? '' : '=' + value.to_s)
    end
  end

  request.body = data
```
